mycroft/configuration/config.py

```diff
@@ -152,3 +157,75 @@ def get_config():
         if not ConfigurationManager._config:
             ConfigurationManager.load()
         return ConfigurationManager._config
+
+
+class ConfigBuilder(object):
```

Comment:
Overall, it seems we can do a simpler code here. The main idea around our configuration management is to read config files in a priority order and replace **the lowest** relevant configs with the **more relevant** ones. From **default** to **user** level in this case. That means we can control an unique array of config files instead of splitting them into categories and adding unnecessary complexity to control those. I'd like to provide a suggestion on top of your changes. @forslund ping me once you have some time, so we can talk about that.